tests/test_url.py

```diff
@@ -420,7 +420,7 @@ def test_safe_url_string_encoding(
     # (%) are not escaped.
     f"a://example.com#{FRAGMENT_TO_ENCODE}",
 }
-if sys.version_info < (3, 11, 4):
+if not(sys.version_info >= (3, 9, 21) or sys.version_info >= (3, 10, 16) or sys.version_info >= (3, 11, 4)):
```

Comment:
I don't think it works this way? 3.10.16 is already >= 3.9.21 (as is 3.10.0).

Reply:
ah right, we need to conditionalize on 3.9/3.10 too. Will update.